packages/keys/src/public-key.ts

```diff
@@ -0,0 +1,19 @@
+import { assertKeyExporterIsAvailable } from '@solana/assertions';
+
+export async function getPublicKeyFromPrivateKey(privateKey: CryptoKey, extractable?: boolean): Promise<CryptoKey> {
+    assertKeyExporterIsAvailable();
+
+    // Export private key.
+    const jwk = await crypto.subtle.exportKey('jwk', privateKey);
+
+    // Remove private data from JWK.
+    delete jwk.d;
+    delete jwk.dp;
+    delete jwk.dq;
+    delete jwk.q;
+    delete jwk.qi;
+    jwk.key_ops = ['verify'];
```

Comment:
Missed setting the `ext` parameter of the jwk according to the value of `extractable`. Might want to write a test to cover mismatches of that nature. My example below handles that in the implementation.

Reply:
I've added some tests.